server/server.go

```diff
@@ -55,6 +55,11 @@ const (
 // Make this a variable so that we can change during tests
 var lameDuckModeInitialDelay = int64(lameDuckModeDefaultInitialDelay)
 
+// In lame duck mode, the server is shutdown and os.Exit() must be called.
+// If this is set to true (in tests), skip the os.Exit() call to prevent
+// test suite to stop.
+var testLameDuckMode = false
```

Comment:
What if two servers running in same process?

Maybe have a server function that sets this scoped to the server instance?

Reply:
Only 1 test is setting this value. But this brings the question if people embed the server and run it in LDM, then the app would exit... so maybe we need to have main.go wait on something that allows then the process to exit, but leave the lameduck function only call Shutdown() and not os.Exit(). Let me explore that avenue..

Reply:
Ok, I took a complete different approach and did a push force. Have another look when you can.